alcs-frontend/src/app/shared/header/header.component.ts

```diff
@@ -12,27 +16,109 @@ import { AuthenticationService } from '../../services/authentication/authenticat
+    private userService: UserService,
     private router: Router
   ) {}
 
   ngOnInit(): void {
-    this.applicationService.$applicationDecisionMakers.subscribe((dms) => {
-      this.decisionMakers = dms;
+    this.authService.isAuthenticated.subscribe((isAuthenticated) => {
+      if (isAuthenticated) {
+        this.currentUser = this.authService.getCurrentUser();
+
+        this.userService.fetchUsers();
+        this.applicationService.setup();
+      }
+    });
+
+    this.userService.$users.subscribe((users) => {
+      this.currentUserProfile = users.find((u) => u.email === this.currentUser.email);
     });
+
+    this.$sortedDecisionMakers = this.applicationService.$applicationDecisionMakers.pipe(
+      map((dms) => this.transformDecisionMakers(dms))
+    );
+  }
+
+  private transformDecisionMakers(dms: ApplicationDecisionMakerDto[]) {
+    this.decisionMakers = dms
+      .map((dm) => {
+        return {
+          ...dm,
+          isFavorite: this.currentUserProfile?.settings?.favoriteBoards?.includes(dm.code),
+        };
+      })
+      .sort((x, y) => this.sortDecisionMakers(x, y));
+    return this.decisionMakers;
+  }
+
+  private sortDecisionMakers(x: ApplicationDecisionMakerDto, y: ApplicationDecisionMakerDto) {
+    if (x.isFavorite && !y.isFavorite) {
+      return -1;
+    }
+
+    if (y.isFavorite && !x.isFavorite) {
+      return 1;
+    }
+
+    if (x.isFavorite === y.isFavorite) {
+      if (x.label > y.label) return 1;
+      if (x.label < y.label) return -1;
+    }
+    return 0;
   }
 
-  onSelectBoard(event: MouseEvent) {
-    const target = event.currentTarget as HTMLInputElement;
-    const boardCode = target.value;
-    this.router.navigateByUrl(`/board/${boardCode}`);
+  onSelectBoard(dmCode: string) {
+    this.router.navigateByUrl(`/board/${dmCode}`);
   }
 
   onLogout() {
     this.authService.logout();
   }
+
+  private updateFavoriteBoardsList(dm: ApplicationDecisionMakerDto) {
+    if (!this.currentUserProfile) {
+      return;
+    }
+
+    if (!this.currentUserProfile?.settings?.favoriteBoards) {
+      this.currentUserProfile.settings = { favoriteBoards: [] };
+    }
+
+    const favoriteBoards = [...this.currentUserProfile.settings.favoriteBoards];
+    if (favoriteBoards.includes(dm.code)) {
+      const index = favoriteBoards.indexOf(dm.code);
+      if (index > -1) {
+        favoriteBoards.splice(index, 1);
+      }
+      this.currentUserProfile.settings.favoriteBoards = favoriteBoards;
+    } else {
+      this.currentUserProfile.settings.favoriteBoards.push(dm.code);
+    }
+  }
+
+  async onFavoriteClicked(event: any, dm: ApplicationDecisionMakerDto) {
+    event.stopPropagation();
+    if (!this.currentUserProfile) {
+      return;
+    }
+
+    this.updateFavoriteBoardsList(dm);
+
+    try {
+      await this.userService.updateUser(this.currentUserProfile);
+      await this.userService.fetchUsers();
+      this.applicationService.$applicationDecisionMakers.next(this.decisionMakers);
```

Comment:
Only the application service should be updating its decision makers. By doing this the decision makers in the application service will not match the decision makers in the observable.